### DISCUSSION TITLE
Neomutt authentication to gmail

### DISCUSSION THREAD

**Discussion by yagarea** [keepassxreboot/keepassxc#6222] [Q&A] [answered]
Hello, is there a way to use keepassxc to authenticate to gmail account in neomutt ?

Thank you for help

**Comment by droidmonkey** [comment, marked as answer]
Use Auto-Type from the gui or keepassxc-cli

**Reply by yagarea** [reply]
how can I use `keepassxc-cli`? Is there a way to send password in to stdout ?

**Reply by droidmonkey** [reply]
Read the man pages please: `man keepassxc-cli`

**Comment by yagarea** [comment]
I solved it:

https://unix.stackexchange.com/questions/638394/keepassxc-cli-send-password-to-stdout/638397#638397

 Thank you for help

**Comment by weilbith** [comment]
Has someone a good idea how I can transform this for a more automated/hidden scenario? So for example I have a systemd user service that makes use of IMAP IDLE to get notified about new mails on the server. Therefore it needs the password to authenticate at the server initially. So far I have the password stored into a file that is encrypted with GnuPG. The command for the password configuration looks like `gpg --for-your-eyes-only --no-tty --batch --quiet --decrypt --recipient key-id /path/to/password/file`. This will automatically bring up the pinentry window to ask for the password or actually for me to insert the smartcard (YubiKey) and provide the PIN to unlock (plus the touch on the key).
But I want to get rid of these encrypted password files and query KeePass directly. But if I use a command like the suggested one above which looks like that: `keepassxc-cli show /path/to/database entry-name --attributes Password --show-protected` it always fails as it requires the password (I don't even have setup the key protection yet) which can't be inserted. 
Any ideas how to solve that? In fact I would just like to get a window to popup which takes care of the user interaction.

**Reply by phoerious** [reply]
I suggest you use the secret service integration.
